### ISSUE TITLE
Accordion 'open/close all sections' cannot be customised or translated

### ISSUE THREAD

**Issue by HughePaul** [alphagov/govuk-frontend#1325]
https://github.com/alphagov/govuk-frontend/blob/1eef89c59c7120050d97cec283d0435cdbb00a45/src/components/accordion/accordion.js#L60

The accordion has hardcoded English text in javascript file, meaning it can't be translated to Welsh. Without being able to translate this our site can't enter Public Beta.

**Comment by 36degrees** [comment]
Hi Paul,

Thanks for raising. This has come up before a few times before – most recently in #1038.

Unfortunately at the minute we don't have a decent solution for localisation / internationalisation.

For now, we suggest that you copy the Javascript for this component into your service's codebase and make the adjustments required for the service you’re working on. I recognise this isn't ideal –  in #1038 I mentioned that we were hoping to look at this properly in 'early [2019],' but we haven't got to it yet.

Hope that helps,

Ollie

**Comment by HughePaul** [comment]
I've penned an MR to try to pull hardcoded text  from data attributes and fall back to the existing English - could this be a suitable route for hardcoded text in the javascript?

**Comment by 36degrees** [comment]
Thanks for doing that! 👏 It sounds like a sensible option, but as it introduces new conventions to the component API we need to spend a little bit of time understanding how it might work for other components – both existing, and new ones – and to make sure it's the right approach.

Our team are currently prioritising work to make [GOV.UK Frontend pass WCAG 2.1 requirements](https://github.com/alphagov/govuk-design-system/issues/677) and we have an external audit coming up soon. Because of this, it may take a little while before we can review your PR.

If this is blocking you, I still recommend copying the JS into your service as the best short-term solution.

Thanks for your understanding.

**Comment by HughePaul** [comment]
Given the new design of the accordion and the predominant English text for "Show all sections", "Hide all sections", "Show" and "Hide" compared to the old component, I think this has become a bit more important.

New accordion on page in welsh:
![new-accordion-welsh](https://user-images.githubusercontent.com/128149/150115593-9a9c6a0b-c3ab-4442-9e9a-47a8c39be3a3.png)

Old accordion on same page:
![old-accordion-welsh](https://user-images.githubusercontent.com/128149/150115598-c1d83134-f87c-4806-8b12-1a362f9037da.png)

For our own components we've been specifying localised text as component args, the nunjucks has been populating `data-text-{name}` attributes in the HTML, and the javascript picks those up from the element. That way the page can populate the localised language text, and the component doesn't need to care about which language it is.

**Comment by owenatgov** [comment]
We ran into a similar problem when developing the [pagination component](https://design-system.service.gov.uk/components/pagination/). Each number originally had a visually hidden "page " suffix which we were struggling to translate without requiring teams to pass complex html. The way we solved this was by instead using `aria-label`'s around the pagination numbers. This resulted in some minor duplication but made the html overall simpler and more importantly, easier to localise. Could we do something like this here?

I can see that @querkmachine has already started to try solving this using the html approach as part of the [localisation work](https://github.com/alphagov/govuk-frontend/pull/2818/files#diff-8f3e70c76932e3229f5a88e95df343e3c98597047d48f5f75fb72ce527f2eb39). That would in theory solve this as well, but I'm sticking my neck out for the `aria-label` approach. We don't need to do that right now and I don't want to disrupt the in-flow i18n work with this.
